actions/lib/action.py

```diff
@@ -12,16 +12,46 @@ def __init__(self, config):
         super(TerraformBaseAction, self).__init__(config)
         self.terraform = Terraform()
 
-    def check_result(self, return_code, stdout, stderr):
+    def check_result(self, return_code, stdout, stderr, return_output=False,
```

Comment:
Need unit tests for new code paths in this function.

Reply:
@nmaludy , I'm really strugging to get the new unit tests to pass. I'm not sure how the other unit test are working in th background. When I try to simulated check_result where it hits the self.terraform.output path. I'm getting:

```FileNotFoundError: [Errno 2] No such file or directory: '/terraform': '/terraform'```

When trying to run the terraform command, it can't find the terraform path.

It's quite confusing because all other tests (e.g. `apply`) use this path, and pass without issues.

https://github.com/StackStorm-Exchange/stackstorm-terraform/blob/master/tests/test_action_apply.py#L21

Reply:
@kingsleyadam it's because those tests mock out the `apply()` function on the terraform client: https://github.com/StackStorm-Exchange/stackstorm-terraform/blob/master/tests/test_action_apply.py#L17

In your case you'll need to mock out (aka patch) the `self.terraform.output()` function

Something like:
```
    @mock.patch("lib.action.Terraform.output")
    def test_check_result(self, mock_output)
        mock_output.return_value = 'blah'
```

Reply:
Sorry for the trouble, I don't have access to the CI build and I'm just going at this a bit blind.

I tried to mock out already, but it's constantly failing when I try to run the method to compare results.

```result = action.check_result(
            test_return_code,
            test_stdout,
            test_stderr,
            return_output=True
        )
```

And it's failing because it's attempting to run `self.terraform.output()` but it can't seem to find a terraform project. I would expect this test to pass is there was a terraform plan at `/terraform`:
```
    def test_check_result_success_with_output(self):
        action = self.get_action_instance({})

        # Set terraform variables for test
        action.terraform.terraform_bin_path = "/usr/bin/terraform"
        action.terraform.working_dir = "/terraform"

        # Declare test input values
        test_return_code = 0
        test_stdout = "Terraform has been successfully initialized!"
        test_stderr = ""

        test_output = dict()
        expected_result = (True, test_output)

        # Execute the run function
        result = action.check_result(
            test_return_code,
            test_stdout,
            test_stderr,
            return_output=True
        )

        # Verify the results
        self.assertEqual(result, expected_result)
```

Reply:
Many trial and error. Your suggestions did the trick.

Thanks for your help. Builds are passing!